engine/flink/executor/src/main/scala/pl/touk/nussknacker/engine/process/util/Serializers.scala

```diff
@@ -66,25 +66,25 @@ object Serializers extends LazyLogging {
       }
     }
 
-    override def read(kryo: Kryo, input: Input, obj: Class[Product]) = {
+    override def read(kryo: Kryo, input: Input, obj: Class[Product]): Product = {
       val constructorParamsCount = input.readInt()
       val constructors = obj.getConstructors
 
       if (constructorParamsCount == 0 && constructors.isEmpty) {
-        Try(EspTypeUtils.companionObject(obj)).recover {
+        Try(EspTypeUtils.companionObject(obj)).recoverWith {
           case e => logger.error(s"Failed to load companion for ${obj.getClass}"); Failure(e)
         }.get
       } else {
         Try({
           val cons = constructors(0)
           val params = (1 to constructorParamsCount).map(_ => kryo.readClassAndObject(input)).toArray[AnyRef]
           cons.newInstance(params: _*).asInstanceOf[Product]
-        }).recover {
+        }).recoverWith {
           case e => logger.error(s"Failed to load obj of class ${obj.getClass.getName}", e); Failure(e)
```

Comment:
please also fix error message here 
```suggestion
          case e => logger.error(s"Failed to load obj of class $obj", e); Failure(e)
```